mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/model/spi/PropertyModel.java

```diff
@@ -6,12 +6,19 @@
  */
 package org.hibernate.search.v6poc.entity.pojo.model.spi;
 
+import java.lang.annotation.Annotation;
+import java.util.stream.Stream;
+
 public interface PropertyModel<T> {
 
 	String getName();
 
 	Class<T> getJavaType();
 
+	<A extends Annotation> Stream<A> getAnnotationsByType(Class<A> annotationType);
+
+	Stream<? extends Annotation> getAnnotationsByMetaAnnotationType(Class<? extends Annotation> metaAnnotationType);
```

Comment:
I'm not sure about it but I have a vague recollection of `Stream#toString()` being unreadable. If so, I must admit that I'm not very fond of using `Stream` as return types.

Reply:
Thing is, it comes really handy when we need to manipulate the "set" of elements, and here we do... a lot. See `org.hibernate.search.v6poc.entity.orm.model.impl.HibernateOrmPropertyModel#getAnnotationsByMetaAnnotationType` (aggregate multiple element sets), `org.hibernate.search.v6poc.entity.orm.model.impl.HibernateOrmIntrospector#getAnnotationsByMetaAnnotationType` (expand some elements to multiple elements, filter the set of elements)...
I guess we could do the filtering "in place", using a `List` or `Set`, but I'm not sure this would be particularly easier to read.

As to debugging... You can just evaluate `<the call that returns a new Stream>.toArray()` and you're good to go. Of course you have to make sure to call to array on a new `Stream` instance, not the one being used in the code, but in this case it doesn't seem to be a problem... ?